api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java

```diff
@@ -9,7 +9,13 @@
 
 public interface CriteriaDao extends CrudRepository<Criteria, Long> {
 
-    @Query("select c from Criteria c where c.type like :type% and c.parentId = :parentId order by c.code asc")
-    List<Criteria> findCriteriaByTypeAndParentId(@Param("type") String type, @Param("parentId") Long parentId);
+    List<Criteria> findCriteriaByTypeAndParentIdOrderByCodeAsc(@Param("type") String type, @Param("parentId") Long parentId);
+
+    @Query("select c from Criteria c " +
+            "where c.type like :type " +
+            "and (lower(c.name) like lower(:name) or lower(c.code) like lower(:code)) " +
```

Comment:
Not sure if it matters for performance, but I believe LIKE searches in MySQL are case-insensitive by default:

https://dev.mysql.com/doc/refman/5.6/en/case-sensitivity.html

So if you are going to use LIKE instead of MATCH (see above), you can probably drop the lower() everywhere in here.

Reply:
my test cases fail if i remove the lower function, but i do like the idea about the fulltext indexing.